lib/jxl/encode.cc

```diff
@@ -557,6 +558,17 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderSetExtraChannelInfo(
   if (index >= enc->metadata.m.num_extra_channels) {
     return JXL_API_ERROR("Invalid value for the index of extra channel");
   }
+  if (!info->exponent_bits_per_sample) {
+    if (!(info->bits_per_sample > 0 && info->bits_per_sample <= 24)) {
+      return JXL_API_ERROR("Invalid value for bits_per_sample");
+    }
+  } else if (!(info->bits_per_sample == 32 &&
+               info->exponent_bits_per_sample == 8) &&
+             !(info->bits_per_sample == 16 &&
+               info->exponent_bits_per_sample == 5)) {
+    return JXL_API_ERROR("Invalid float description");
```

Comment:
This condition is overly strict. This checks if it is binary32 or binary16, which are the two cases we have a pixelformat for. But the bitstream allows anything with at most 8 exp bits and at most 24 remaining bits (mantissa+sign).

Reply:
This is the behavior that the code had before this MR as well (the encoder was cautious), perhaps changing behavior could be done in another one?

Reply:
OK to split it up.

Reply:
Sure I can split it into two pull requests.

Reply:
first part here, leaving old behavior